File: src/main/java/org/alham/alhamfirst/service/user/UserService.java
package org.alham.alhamfirst.service.user;


import org.alham.alhamfirst.dto.user.UserDTO;
import org.alham.alhamfirst.entity.User;

import java.util.Optional;

public interface UserService {

    public UserDTO getUser(long userIdx);
    public User createUser(UserDTO userDTO);
    public UserDTO getUserByIdx(long userId);
    public UserDTO getUserByEncryptedId(String encryptedId);


}

File: src/main/java/org/alham/alhamfirst/service/user/UserService.kt
package org.alham.alhamfirst.service.user;


import org.alham.alhamfirst.domain.dto.user.UserDTO;
import org.alham.alhamfirst.domain.entity.User;

interface UserService {

    fun getUser(userIdx: Long) : UserDTO
    fun createUser(userDTO: UserDTO): User
    fun getUserByIdx(userId: Long): UserDTO
    fun getUserByEncryptedId(encryptedId: String): UserDTO;


}
